src/__tests__/cdfDatasource.test.ts

```diff
@@ -74,13 +69,15 @@ describe('CDF datasource', () => {
 
   describe('promiser', () => {
     it('should return failures and successes', async () => {
-      const queries = [0, 1, 2, 3];
-      const metadatas = ['a', 'b', 'c', 'd'];
+      const queries = [0, 1, 2, 3] as unknown as CDFDataQueryRequest[];
+      const metadatas = ['a', 'b', 'c', 'd'] as unknown as ResponseMetadata[];
```

Comment:
this is very bad

Reply:
Yes it's a bad test. We will not refactor the tests as part of the port to React. At least this shows how bad it already was.

Reply:
the function used to be generic.
so it worked with anything before,
but then you guys decided to lock it to this implementation :)

Reply:
Aha, yes that's right. I still think it's a good choice. The `promiser` function should not need to care about Query, Metadata or Responses so we should refactor this. It's too much detail for a function that simply takes a list of promise generating functions and returns a single promise with accumulated result.

Reply:
exactly!

Reply:
I will try to refactor this